generative/networks/blocks/encoder_modules.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from functools import partial
+
+import torch
+import torch.nn as nn
+from monai.networks.blocks import Convolution
+
+__all__ = ["SpatialRescaler"]
+
+
+class SpatialRescaler(nn.Module):
+    """
+    SpatialRescaler based on https://github.com/CompVis/latent-diffusion/blob/main/ldm/modules/encoders/modules.py
+
+    Args:
+        n_stages: number of interpolation stages.
+        method: algorithm used for sampling.
+        multiplier: multiplier for spatial size. If scale_factor is a tuple,
+            its length has to match the number of spatial dimensions.
+        in_channels: number of input channels.
+        out_channels: number of output channels.
+        bias: whether to have a bias term.
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int = 2,
+        n_stages: int = 1,
+        method: str = "bilinear",
+        multiplier: float = 0.5,
```

Comment:
in addition to a multiplier option I think it would be good to give users the option to specify the output size they desire (that is, unless i'm missing the intended use of this encoder block?).


this should be possible by allowing users to use the `size` argument in `torch.nn.functional.interpolate`

Reply:
Agree, added `size` argument support and corresponding tests.